cmd/lingo/main.go

```diff
@@ -112,6 +115,8 @@ func run() error {
 	le := leader.NewElection(clientset, hostname, namespace)
 
 	queueManager := queue.NewManager(concurrencyPerReplica)
+	metricsRegistry := prometheus.WrapRegistererWithPrefix("lingo_", metrics.Registry)
```

Comment:
Note: lingo actually already has a metrics exporter running. This is done in `ctrl.NewManager()` inside of `main()`, which exposes metrics about the kubernetes controller functionality. The actual package that starts the metrics server is [here](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/metrics/server). Here is a guide on how to expose additional metrics: https://book.kubebuilder.io/reference/metrics ... Thoughts on going that direction?

Reply:
PS: You can see these metrics by running lingo and curling:

```
curl localhost:8082/metrics | head
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 22737    0 22737# HELP certwatcher_read_certificate_errors_total Total number of certificate read errors
 # TYPE certwatcher_read_certificate_errors_total counter
 certwatcher_read_certificate_errors_total 0
 # HELP certwatcher_read_certificate_total Total number of certificate reads
# TYPE certwatcher_read_certificate_total counter
 certwatcher_read_certificate_total 0
0# HELP controller_runtime_active_workers Number of currently used workers per controller
 # TYPE controller_runtime_active_workers gauge
 controller_runtime_active_workers{controller="configmap"} 0
 controller_runtime_active_workers{controller="deployment"} 0
  0  2079k      0 --:--:-- --:--:-- --:--:-- 21.6M
```

Reply:
Thanks for the link! I am using the global `metrics.Registry` as suggested in the book. The new metrics are returned by this endpoint already. It can be confusing but `WrapRegistererWithPrefix` is just some short cut to have the `lingo_` prefix attached to all custom metric names only. I don't have a strong opinion on the naming.